projects/kit/components/calendar-range/calendar-range.component.ts

```diff
@@ -186,6 +183,7 @@ export class TuiCalendarRange implements OnInit, OnChanges {
             this.selectedActivePeriod = item;
             this.updateValue(item.range.dayLimit(this.min, this.max));
             this.itemChange.emit(item);
+            this.defaultViewedMonth = this.value?.to ?? this.defaultViewedMonth;
```

Comment:
Double code with
https://github.com/taiga-family/taiga-ui/pull/8843/files#diff-8fda4881ec9c2cd09a2bf32685bee1e50c9f4880e0fb41586f6e88bc0001e3e7R99

should me carried out to the one general private method

Reply:
I think we can use existing method https://github.com/taiga-family/taiga-ui/blob/main/projects/kit/components/calendar-range/calendar-range.component.ts#L252-L260 

it's important to check `this.items.length` due to show end of period when items > 0 and starts of period, when it's default calendar without items

Reply:
```suggestion
            this.initDefaultViewedMonth();
```

I think we should it to the end of method `onItemSelect`, not only in `if` branch